pkg/controller/secretsmanager/controller.go

```diff
@@ -0,0 +1,287 @@
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotSecret)
+	}
+
+	cr.SetConditions(runtimev1.Creating())
+
+	secret, err := secretsmanager.GetSecret(ctx, e.kube, cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	_, err = e.client.CreateSecretRequest(secretsmanager.GenerateCreateSecretsmanagerInput(meta.GetExternalName(cr), &cr.Spec.ForProvider, secret)).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotSecret)
+	}
+
+	req, err := e.client.DescribeSecretRequest(&awssecretsmanager.DescribeSecretInput{SecretId: awscommon.String(meta.GetExternalName(cr))}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errDescribeSecretFailed)
+	}
+
+	err = e.updateTags(ctx, cr, req)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	secret, err := secretsmanager.GetSecret(ctx, e.kube, cr)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	_, err = e.client.UpdateSecretRequest(secretsmanager.GenerateUpdateSecretInput(meta.GetExternalName(cr), cr.Spec.ForProvider, secret)).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errUpdateFailed)
+	}
+
+	if cmp.Equal(v1alpha1.GenerateSecretTags(cr.Spec.ForProvider.Tags), req.Tags) {
+		_, err = e.client.TagResourceRequest(&awssecretsmanager.TagResourceInput{
+			SecretId: awscommon.String(meta.GetExternalName(cr)),
+			Tags:     v1alpha1.GenerateSecretTags(cr.Spec.ForProvider.Tags),
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errTagResourceRequest)
+		}
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return errors.New(errNotSecret)
+	}
+
+	cr.SetConditions(runtimev1.Deleting())
+
+	if cr.Status.AtProvider.DeletionDate != nil {
+		// only request a new deletion if the user has changed the recovery window
+		r := secretsmanager.GetRecoveryWindowInDays(cr.Spec.ForProvider.RecoveryWindowInDays)
+		oldDeletionDate := cr.Status.AtProvider.DeletionDate
+		newDeletionDate := &metav1.Time{Time: cr.Status.AtProvider.DeletedDate.AddDate(0, 0, r)}
+		if oldDeletionDate.Equal(newDeletionDate) {
+			return nil
+		}
+	}
+
+	rsp, err := e.client.DeleteSecretRequest(secretsmanager.GenerateDeleteSecretInput(meta.GetExternalName(cr), cr.Spec.ForProvider)).Send(ctx)
+	if err != nil {
+		return errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errDeleteFailed)
+	}
+	cr.Status.AtProvider.DeletionDate = &metav1.Time{Time: *rsp.DeletionDate}
```

Comment:
nitpick: this will be assigned in `Observe` anyway.

Reply:
Indeed, however, only after trying to retrieve the secret value from AWS. We experienced an error when doing so with a deleted secret. For that reason, we make sure that `DeletionDate`is added here, so during the next observation, the secret is skipped and the error does not occur.

Reply:
`DeletedDate` is set in `Observe`. `DeletionDate` is only returned by the `DeleteSecretRequest` call.